From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eneas U de Queiroz <cotequeiroz@gmail.com>
Date: Tue, 9 Mar 2021 16:56:13 -0300
Subject: Openwrt: don't search for boost libs in host dirs

While searching for the boost_system library in boost.m4, configure
tries to find boost_system-mt before boost_system.  The presence of
boost_system-mt in the staging dir depends on
CONFIG_boost-use-name-tags.  If it is not defined (default), and there
is a boost_system-mt library in the host system, it will be used, and
the build will fail.

This patch removes the host paths from the search loop, preserving the
rest of the detection logic.

Alternatively, boost_cv_lib_context_LIBS could be used to avoid library
detection code entirely, but then the mt- variant would never be used.

Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>

--- a/m4/boost.m4
+++ b/m4/boost.m4
@@ -467,8 +467,7 @@ for boost_rtopt_ in $boost_rtopt '' -d;
     boost_tmp_lib=$with_boost
     test x"$with_boost" = x && boost_tmp_lib=${boost_cv_inc_path%/include}
     for boost_ldpath in "$boost_tmp_lib/lib" '' \
-             /opt/local/lib* /usr/local/lib* /opt/lib* /usr/lib* \
-             "$with_boost" C:/Boost/lib /lib*
+             "$with_boost"
     do
       # Don't waste time with directories that don't exist.
       if test x"$boost_ldpath" != x && test ! -e "$boost_ldpath"; then
